nx/lib/nx.ex

```diff
@@ -7183,6 +7184,26 @@ defmodule Nx do
           [6, 13, 21]
         ]
       >
+
+      iex> Nx.cumulative_sum(Nx.iota({3, 3}), axis: 0, reverse: true)
+      #Nx.Tensor<
+        s64[3][3]
+        [
+          [6, 7, 8],
+          [9, 11, 13],
+          [9, 12, 15]
+        ]
+      >
+
+      iex> Nx.cumulative_sum(Nx.iota({3, 3}), axis: 1, reverse: true)
+      #Nx.Tensor<
+        s64[3][3]
+        [
+          [2, 3, 3],
```

Comment:
```elixir
iex> [0, 1, 2] |> Explorer.Series.from_list() |> Explorer.Series.cumulative_sum(reverse: true)
#Explorer.Series<
  integer[3]
  [3, 3, 2]
>
```

Reverse only determines the direction we go. The result is equivalent to `reverse + cumulative_* + reverse`.

The default windowed implementation could be optimised by padding right rather than left, but for torch I think we will need the double reverse.

Reply:
Neither numpy nor pytorch support this option, but [tf.math.cumsum](https://www.tensorflow.org/api_docs/python/tf/math/cumsum) does.